src/tigerbeetle/inspect.zig

```diff
@@ -225,6 +231,49 @@ fn inspect_constants(output: std.io.AnyWriter) !void {
     }
 }
 
+fn inspect_metrics(output: std.io.AnyWriter) !void {
+    const EventMetricTag = std.meta.Tag(EventMetric);
+    const EventTimingTag = std.meta.Tag(EventTiming);
+
+    const stats_per_gauge = std.meta.fields(EventMetricAggregate).len - 1; // -1 to ignore `event`.
+    const stats_per_timing = std.meta.fields(std.meta.FieldType(EventTimingAggregate, .values)).len;
+    var stats_total: usize = 0;
+
+    log.info("Format: [metric type]: [metric name]([metric tags])=[metric cardinality]", .{});
+
+    inline for (std.meta.fields(EventMetric)) |field| {
+        const metric_tag = std.meta.stringToEnum(EventMetricTag, field.name).?;
+        try output.print("gauge: {s}(", .{field.name});
+        if (field.type != void) {
+            inline for (std.meta.fields(field.type), 0..) |data_field, i| {
+                if (i != 0) try output.print(", ");
+                try output.print("{s}", .{data_field.name});
+            }
+        }
+        const metric_stats = EventMetric.slot_limits.get(metric_tag) * stats_per_gauge;
+        try output.print(")={}\n", .{metric_stats});
+        stats_total += metric_stats;
+    }
+    inline for (std.meta.fields(EventTiming)) |field| {
+        const timing_tag = std.meta.stringToEnum(EventTimingTag, field.name).?;
+        try output.print("timing: {s}(", .{field.name});
+        if (field.type != void) {
+            inline for (std.meta.fields(field.type), 0..) |data_field, i| {
+                if (i != 0) try output.print(", ");
+                try output.print("{s}", .{data_field.name});
+            }
+        }
+        const timing_stats = EventTiming.slot_limits.get(timing_tag) * stats_per_timing;
+        try output.print(")={}\n", .{timing_stats});
+        stats_total += timing_stats;
+    }
+    log.info("Total stats per replica: {}", .{stats_total});
+    log.info(
+        "(All metrics are tagged with the replica, so the cluster has 6x as many metrics.)",
+        .{},
```

Comment:
Hmmm - do we call them metrics or stats? Or is a stat a metric + specific tags?

Reply:
Afaik there isn't any consistent terminology, but I try to use `stat = metric + tags`. So by that definition I guess this should be "stats"!